pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -176,14 +176,59 @@ func TestPriorityQueue_AddWithReversePriorityLessFunc(t *testing.T) {
 	}
 }
 
+func TestInFlightPods(t *testing.T) {
```

Comment:
I would like to see a test table for this.

I think I already spotted some bugs in the implementation that should be covered by unit test.